packages/@aws-cdk/aws-eks/lib/kubectl-provider.ts

```diff
@@ -106,6 +106,7 @@ export class KubectlProvider extends NestedStack {
       onEventHandler: handler,
       vpc: cluster.kubectlPrivateSubnets ? cluster.vpc : undefined,
       vpcSubnets: cluster.kubectlPrivateSubnets ? { subnets: cluster.kubectlPrivateSubnets } : undefined,
+      securityGroups: cluster.kubectlPrivateSubnets ? [cluster.kubectlSecurityGroup!] : undefined,
```

Comment:
Replace `!` with an explicit if/throw please.

Reply:
Its already covered here:

https://github.com/aws/aws-cdk/blob/8f1cfe4087b765cecd3ceb01745167646d617fec/packages/%40aws-cdk/aws-eks/lib/kubectl-provider.ts#L66-L68

And I only want to pass it if `cluster.kubectlPrivateSubnets` is defined. Did you have something else in mind?

Reply:
Just as a rule of thumb I always prefer an explicit if/throw over `!`. If this assertion fails the user experience is horrible.

Reply:
Yeah I agree. But the assertion you're talking about already exists - it just doesn't satisfy the compiler because of the extra condition. I changed it to `securityGroups: cluster.kubectlSecurityGroup ? [cluster.kubectlSecurityGroup] : undefined` which will have the same affect.